### ISSUE TITLE
Multiple ways to initialise machine context

### ISSUE THREAD

**Issue by JoshuaAmaju** [statelyai/xstate#2860]
The doc currently shows one way to initialise the context object of a machine. And it appears there are other ways of doing that not included in the docs. Currently what we have is:

```javascript
...
context: {
      amount: 0
}
...
```

mean while this is also possible:

```javascript
context: () => {
    return {
        amount: 0
    };
}
```

**Comment by josmithua** [comment]
Two other ways, if you already have a machine instance, are:

```js
const machineWithContext = yourMachine.withContext(context)
```

and 

```js
const machineWithContext = yourMachine.withContext(() => getContext())
```

If `yourMachine` already has existing context, the new context from the call to `withContext` will be shallow merged with the old context.

**Comment by ShaswatPrabhat** [comment]
Does this need an update of the the documentation? I can give it a shot if open.

**Comment by davidkpiano** [comment]
> Does this need an update of the the documentation? I can give it a shot if open.

Yes, that would be great.

This pattern: `context: () => ...` is for _lazy_ context; that is, context that you don't want to create until the machine is actually created/used.

**Comment by with-heart** [comment]
> Not sure if we need a changeset for doc changes though.

Nope! Changesets are only for changes that users of XState packages would want to know about. Changes to the docs, tooling, CI stuff, etc. don't get a changeset.

**Comment by ShaswatPrabhat** [comment]
Hi, can someone please help review the changes. I was looking to add this contribution to my hacktoberfest credits if possible.

Thanks in advance.
